Dockerfile-dind-rootless.template

```diff
@@ -0,0 +1,70 @@
+FROM docker:%%VERSION%%-dind
+
+# busybox "ip" is insufficient:
+#   [rootlesskit:child ] error: executing [[ip tuntap add name tap0 mode tap] [ip link set tap0 address 02:50:00:00:00:01]]: exit status 1
+RUN apk add --no-cache iproute2
+
+# "/run/user/UID" will be used by default as the value of XDG_RUNTIME_DIR
+RUN mkdir /run/user && chmod 1777 /run/user
+
+# create a default user preconfigured for running rootless dockerd
+RUN set -eux; \
+	adduser -h /home/rootless -g 'Rootless' -D -u 1000 rootless; \
```

Comment:
the UID should be configurable as `ARG` https://github.com/docker-library/docker/pull/165/files#diff-3a278e88442db689fbd14924162365f4R12

Reply:
Why?  Folks shouldn't have to / shouldn't be rebuilding the image (I think the usability of that is way too low) -- the way it's configured, setting a reasonable value for `HOME` and setting up `/etc/subuid` and `/etc/subgid` *should* be enough for things to work (although some of the tools in use might also require `/etc/passwd` to be updated, but these are all pretty normal things for running a container as an "arbitrary user" and avoids requiring users to build their own custom images).

Reply:
the UID 1000 is likely to overwrap with other users on the host, and it might break the user's env when the daemon got compromised.

Reply:
Also the number of subuids should be reconfigurable, as some user might want more than 65536 subuids

Reply:
The goal here is that the user can do something as trivial as the following in order to customize their environment (and thus adjust the numbers to their heart's content):

```dockerfile
FROM docker:19.03-dind-rootless
USER root
RUN adduser -h /home/user -D -u 1234 user
RUN echo 'user:101234:65536' >> /etc/subuid
RUN echo 'user:101234:65536' >> /etc/subgid
USER user
```

(most of this is also easily accomplishable via a `docker run` line, although a bit more gymnastics are involved)